File: machinery/studio/frontend/src/types/audit.ts
export type AuditVerdict =
  | 'not_ready'
  | 'promising_not_ready'
  | 'narrowly_usable'
  | 'ready_for_limited_users'
  | 'product_ready'

export type AuditRunStatus = 'pending' | 'running' | 'complete' | 'failed'
export type AuditSubagentStatus = 'pass' | 'warn' | 'fail' | 'blocked'
export type FindingSeverity = 'critical' | 'high' | 'medium' | 'low'
export type EvidenceKind = 'file' | 'command' | 'test' | 'doc' | 'api' | 'ui'

export interface EvidenceRef {
  id: string
  kind: EvidenceKind
  path?: string | null
  command?: string | null
  observed: string
}

export interface Finding {
  severity: FindingSeverity
  claim: string
  evidence_refs: string[]
  product_risk: string
  recommended_next_step: string
}

export interface AuditSubagentResult {
  id: string
  name: string
  category: string
  status: AuditSubagentStatus
  score: number
  max_score: number
  findings: Finding[]
  evidence: EvidenceRef[]
  open_questions: string[]
}

export interface ReadinessReport {
  one_sentence_product: string
  specific_user: string
  works_today_without_founder: string[]
  breaks_first: string[]
  next_risk_reducing_milestone: string
  verdict: AuditVerdict
  category_scores: Record<string, number>
  executive_summary: string
  highest_risk_assumption: string
  priority_findings: Finding[]
}

export interface AuditRun {
  id: string
  created_at: string
  repo_root: string
  persona: 'burned_out_bay_area_engineer'
  frontend_framework: 'react'
  mode: 'read_only'
  target: 'texgraph_current_system'
  status: AuditRunStatus
  subagents: AuditSubagentResult[]
  report: ReadinessReport | null
}

File: backend/api/app/models/audit.py
"""Pydantic v2 models for the product readiness audit orchestrator."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


AuditVerdict = Literal[
    "not_ready",
    "promising_not_ready",
    "narrowly_usable",
    "ready_for_limited_users",
    "product_ready",
]
AuditRunStatus = Literal["pending", "running", "complete", "failed"]
SubagentStatus = Literal["pass", "warn", "fail", "blocked"]
FindingSeverity = Literal["critical", "high", "medium", "low"]
EvidenceKind = Literal["file", "command", "test", "doc", "api", "ui"]


class EvidenceRef(BaseModel):
    id: str
    kind: EvidenceKind
    path: str | None = None
    command: str | None = None
    observed: str


class Finding(BaseModel):
    severity: FindingSeverity
    claim: str
    evidence_refs: list[str]
    product_risk: str
    recommended_next_step: str


class AuditSubagentResult(BaseModel):
    id: str
    name: str
    category: str
    status: SubagentStatus
    score: int = Field(..., ge=0)
    max_score: int = Field(default=10, ge=1)
    findings: list[Finding] = []
    evidence: list[EvidenceRef] = []
    open_questions: list[str] = []


class ReadinessReport(BaseModel):
    one_sentence_product: str
    specific_user: str
    works_today_without_founder: list[str]
    breaks_first: list[str]
    next_risk_reducing_milestone: str
    verdict: AuditVerdict
    category_scores: dict[str, int]
    executive_summary: str
    highest_risk_assumption: str
    priority_findings: list[Finding]


class AuditRun(BaseModel):
    id: str
    created_at: str
    repo_root: str
    persona: Literal["burned_out_bay_area_engineer"] = "burned_out_bay_area_engineer"
    frontend_framework: Literal["react"] = "react"
    mode: Literal["read_only"] = "read_only"
    target: Literal["texgraph_current_system"] = "texgraph_current_system"
    status: AuditRunStatus
    subagents: list[AuditSubagentResult]
    report: ReadinessReport | None = None
